python/ray/tune/tests/test_tuner_restore.py

```diff
@@ -746,6 +757,66 @@ def test_tuner_restore_from_moved_experiment_path(
     assert not old_local_dir.exists()
 
 
+def test_tuner_restore_from_moved_cloud_uri(
+    ray_start_2_cpus, tmp_path, clear_memory_filesys
+):
+    """Test that that restoring an experiment that was moved to a new remote URI
+    resumes and continues saving new results at that URI."""
+
+    def failing_fn(config):
+        data = {"score": 1}
+        session.report(data, checkpoint=Checkpoint.from_dict(data))
+        raise RuntimeError("Failing!")
+
+    tuner = Tuner(
+        failing_fn,
+        run_config=RunConfig(
+            name="exp_dir",
+            local_dir=str(tmp_path / "ray_results"),
+            sync_config=tune.SyncConfig(upload_dir="memory:///original"),
+        ),
+        tune_config=TuneConfig(trial_dirname_creator=lambda _: "test"),
+    )
+    tuner.fit()
+
+    # mv memory:///original/exp_dir memory:///moved/new_exp_dir
+    download_from_uri(
+        "memory:///original/exp_dir", str(tmp_path / "moved" / "new_exp_dir")
+    )
+    delete_at_uri("memory:///original")
+    upload_to_uri(str(tmp_path / "moved"), "memory:///moved")
+
+    tuner = Tuner.restore("memory:///moved/new_exp_dir", resume_errored=True)
+    # This is needed because the mock memory:/// filesystem doesn't
+    # sync checkpoints properly, so this just copies the original checkpoint
+    # to the new local directory.
+    # NOTE: A new local directory is used since the experiment name got modified.
```

Comment:
Can you elaborate what it doesn't do?
Is this just specific to the memory fs or does this come up with other fs as well?

Reply:
Just memory fs. I think the problem is that the trainable actor uploads its checkpoints to a different `memory://` filesystem that's not accessible by the main process. Then, when I download from the URI, only files uploaded by the driver are there.